src/components/CodeView/index.tsx

```diff
@@ -68,17 +69,24 @@ export class CodeViewBase extends React.Component<Props> {
 
     const language = getLanguageFromMimeType(mimeType);
 
+    const globalLinterMessages = selectedMessageMap
+      ? selectedMessageMap.global.map((message) => {
+          return <LinterMessage key={message.uid} message={message} />;
+        })
+      : [];
+
     return (
       <React.Fragment>
-        {selectedMessageMap &&
-          selectedMessageMap.global.length > 0 &&
-          isLineSelected(getCodeLineAnchorID(0), location) && (
-            <div id={getCodeLineAnchorID(0)} ref={_scrollToSelectedLine} />
-          )}
-        {selectedMessageMap &&
-          selectedMessageMap.global.map((message) => {
-            return <LinterMessage key={message.uid} message={message} />;
-          })}
+        <GloablLinterMessages
+          scrollToSelectedLine={
+            isLineSelected(getCodeLineAnchorID(0), location)
+              ? _scrollToSelectedLine
+              : undefined
+          }
+        >
+          {globalLinterMessages}
```

Comment:
Rather than make the caller create each `<LinterMessage>` component, it would save some duplication to accept a list of messages, like:

```jsx
<GlobalLinterMessages messages={selectedMessageMap.global} />
```

With that, `GlobalLinterMessages` can be responsible for mapping each array item to `<LinterMessage>`.